routers/api/v1/repo/find_files.go

```diff
@@ -0,0 +1,114 @@
+// Copyright 2021 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package repo
+
+import (
+	"fmt"
+	"net/http"
+	"regexp"
+	"strings"
+
+	"code.gitea.io/gitea/modules/base"
+	"code.gitea.io/gitea/modules/context"
+	"code.gitea.io/gitea/modules/git"
+
+	"github.com/go-enry/go-enry/v2"
+)
+
+var excludedDirs = []string{
+	"vendor",
+	"build",
+	"log",
+	"tmp",
+}
+
+// GetRepoFiles get all files' entries of a repository
+func GetRepoFiles(ctx *context.APIContext) {
```

Comment:
It seems not worth to make it as an API. Once it becomes an API, the changing will be very difficult.

(some recent PRs also refactored some AJAX requests from API to internal AJAX endpoint only)

If the AJAX request is only used by a page, it could be put under the `/org/repo/....` , just like what GitHub does (`https://github.com/go-gitea/gitea/tree-list/62c74d215b3ba5dc34841268c5b9e09bc4531804`)